### PR TITLE
NDRS-763: Cleanup of `linear_chain_sync.rs` 

### PR DESCRIPTION
**Pull request by goral09**
Small clean-up of `lienar_chain_sync.rs` file:
* Extracts `ReactorEventT` trait to a file,
* Extracts `State` struct to a file.
* Wraps peer-related information with a `PeersState` struct and extracts it to a file.
* Adds `peer` to a `DeploysNotFound` result (follows the pattern of other results in the component).
* Adds `GetDeploysResult` enum variant to follow the pattern used when downloading blocks.

### Recommend reviewing commit-by-commit.

### CODE REVIEW

File: node/src/components/linear_chain_sync.rs [hunk 1]
```diff
@@ -542,31 +395,32 @@ where
...
+                    effect_builder.execute_block(finalized_block).ignore()
                 }
-                Some(peer) => {
-                    let block_hash = (*block_header).hash();
-                    trace!(%block_hash, next_peer=%peer,
-                        "failed to download deploys from a peer. Trying next one");
-                    fetch_block_deploys(effect_builder, peer, *block_header)
+                event::DeploysResult::NotFound(block_header, peer) => {
+                    let block_hash = block_header.hash();
+                    trace!(%block_hash, %peer, "deploy for linear chain block not found. Trying next peer");
```
Comment (afck) [inline node/src/components/linear_chain_sync.rs:410]: Line too long.

File: node/src/components/linear_chain_sync.rs [hunk 2]
```diff
@@ -542,31 +395,32 @@ where
...
+                    match self.peers.random_peer() {
+                        None => {
+                            let block_hash = block_header.hash();
+                            error!(%block_hash, "could not download deploys from linear chain block.");
```
Comment (afck) [inline node/src/components/linear_chain_sync.rs:413]: Line too long.

File: node/src/components/linear_chain_sync/peers.rs [hunk 3]
```diff
@@ -0,0 +1,54 @@
+use datasize::DataSize;
+use rand::{seq::SliceRandom, Rng};
+
+#[derive(DataSize, Debug)]
+pub struct PeersState<I> {
+    // Set of peers that we can requests block from.
```
Comment (afck) [inline node/src/components/linear_chain_sync/peers.rs:6]:
```suggestion
    /// Set of peers that we can request blocks from.
```

File: node/src/components/linear_chain_sync/peers.rs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    peers: Vec<I>,
+    // Peers we have not yet requested current block from.
+    // NOTE: Maybe use a bitmask to decide which peers were tried?.
```
Comment (afck) [inline node/src/components/linear_chain_sync/peers.rs:9]:
```suggestion
    // NOTE: Maybe use a bitmask to decide which peers were tried?
```

File: node/src/components/linear_chain_sync.rs [hunk 5]
```diff
@@ -542,31 +395,32 @@ where
...
-                    let block_hash = (*block_header).hash();
-                    trace!(%block_hash, next_peer=%peer,
-                        "failed to download deploys from a peer. Trying next one");
-                    fetch_block_deploys(effect_builder, peer, *block_header)
+                event::DeploysResult::NotFound(block_header, peer) => {
+                    let block_hash = block_header.hash();
+                    trace!(%block_hash, %peer, "deploy for linear chain block not found. Trying next peer");
+                    match self.peers.random_peer() {
+                        None => {
+                            let block_hash = block_header.hash();
```
Comment (Fraser999) [inline node/src/components/linear_chain_sync.rs:413]: Do we need this line?  We already have the `block_hash` here I think.

**Review by Fraser999 — APPROVED** [review]
One non-blocking nitpick.  Otherwise LGTM.

File: node/src/components/linear_chain_sync.rs (hunk 5 above)
Comment (goral09) [reply node/src/components/linear_chain_sync.rs:413]: Right, we shouldn't need it. Good catch.

Comment (goral09) [reply node/src/components/linear_chain_sync.rs]: Done in [`a9dcbf7` (#783)](https://github.com/CasperLabs/casper-node/pull/783/commits/a9dcbf708b8cfd18f5ca74e13d958615ac5dc9f1)

**Review by afck — APPROVED** [review]
Looks good; just a few nit-picks.

File: node/src/components/linear_chain_sync.rs (hunk 1 above)
Comment (goral09) [reply node/src/components/linear_chain_sync.rs:410]: How do you know it's too long? Are you counting chars?

File: node/src/components/linear_chain_sync/peers.rs (hunk 4 above)
Comment (goral09) [reply node/src/components/linear_chain_sync/peers.rs]: Done in [`77e0123` (#783)](https://github.com/CasperLabs/casper-node/pull/783/commits/77e01237332b715fb645ee57771ed17afff9c7e9)

File: node/src/components/linear_chain_sync/peers.rs (hunk 3 above)
Comment (goral09) [reply node/src/components/linear_chain_sync/peers.rs]: Done in [`77e0123` (#783)](https://github.com/CasperLabs/casper-node/pull/783/commits/77e01237332b715fb645ee57771ed17afff9c7e9)

File: node/src/components/linear_chain_sync.rs (hunk 2 above)
Comment (goral09) [reply node/src/components/linear_chain_sync.rs]: Done in [`77e0123` (#783)](https://github.com/CasperLabs/casper-node/pull/783/commits/77e01237332b715fb645ee57771ed17afff9c7e9)

File: node/src/components/linear_chain_sync.rs (hunk 1 above)
Comment (goral09) [reply node/src/components/linear_chain_sync.rs]: Done in [`77e0123` (#783)](https://github.com/CasperLabs/casper-node/pull/783/commits/77e01237332b715fb645ee57771ed17afff9c7e9)

File: node/src/components/linear_chain_sync.rs [hunk 6]
```diff
@@ -397,8 +250,8 @@ impl<I: Clone + PartialEq + 'static> LinearChainSync<I> {
         I: Send + 'static,
         REv: ReactorEventT<I>,
     {
-        self.reset_peers(rng);
-        let peer = self.random_peer_unsafe();
+        self.peers.reset_peers(rng);
+        let peer = self.peers.random_peer_unsafe();
```
[the remark this answers is not in the document]
Comment (goral09) [reply node/src/components/linear_chain_sync.rs]: Done in [`77e0123` (#783)](https://github.com/CasperLabs/casper-node/pull/783/commits/77e01237332b715fb645ee57771ed17afff9c7e9)

**Comment by goral09** [comment]
bors merge

File: node/src/components/linear_chain_sync.rs (hunk 1 above)
Comment (afck) [reply node/src/components/linear_chain_sync.rs]: > Are you counting chars?

I haven an eye for that. If it seems too long I paste it into an editor and that usually confirms my suspicion. :sunglasses:

**Pull request merged**